packages/expo-splash-screen/src/index.native.ts

```diff
@@ -1,9 +1,8 @@
 import { requireOptionalNativeModule } from 'expo-modules-core';
 
-const SplashModule = requireOptionalNativeModule('ExpoSplashScreen') as {
-  preventAutoHideAsync: () => Promise<boolean>;
-  hideAsync: () => Promise<boolean>;
-} | null;
+import { SplashScreenNativeModule, SplashScreenOptions } from './SplashScreen.types';
+
+const SplashModule = requireOptionalNativeModule<SplashScreenNativeModule>('ExpoSplashScreen');
```

Comment:
```suggestion
const SplashModule = requireNativeModule<SplashScreenNativeModule>('ExpoSplashScreen');
```

Maybe you want to fail fast if the native code is not there?